### ISSUE TITLE
`rename` op in `clickhouse` dialect not recognized 

### ISSUE THREAD

**Issue by maciejskorski** [sqlfluff/sqlfluff#6071]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Consider the following `test.sql` file
```SQL
RENAME TABLE my_db.my_table TO my_db.my_new_table
```
Invoking linting under version 2.3.5
```bash
sqlfluff lint --dialect clickhouse test.sql
```
results in the error
```bash
== [test.sql] FAIL                                                                                                                                                   
L:   6 | P:   1 |  PRS | Line 6, Position 1: Found unparsable section: 'RENAME
                       | TABLE my_db.my_table TO my_db.my_...'
WARNING: Parsing errors found and dialect is set to 'clickhouse'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Expected Behaviour

Handle it correctly.

### Observed Behaviour

Valid syntax not recognized.

### How to reproduce

Consider the following `test.sql` file
```SQL
RENAME TABLE my_db.my_table TO my_db.my_new_table
```
Invoking linting under version 2.3.5
```bash
sqlfluff lint --dialect clickhouse test.sql
```
results in the error
```bash
== [test.sql] FAIL                                                                                                                                                   
L:   6 | P:   1 |  PRS | Line 6, Position 1: Found unparsable section: 'RENAME
                       | TABLE my_db.my_table TO my_db.my_...'
WARNING: Parsing errors found and dialect is set to 'clickhouse'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Dialect

clickhouse

### Version

2.3.5

### Configuration

```
[sqlfluff]

dialect = clickhouse
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
